storage/flush.go

```diff
@@ -128,7 +118,7 @@ func (m *flushManager) namespaceFlushTimes(ns databaseNamespace, curr time.Time)
 
 	candidateTimes := timesInRange(earliest, latest, blockSize)
 	return filterTimes(candidateTimes, func(t time.Time) bool {
-		return ns.NeedsFlush(t, t.Add(blockSize))
+		return ns.NeedsFlush(t, t)
```

Comment:
`databaseNamespace.NeedsFlush(...)` is inclusive on both ends; so we only need to check `NeedsFlush(t, t)` to determine if the block at time `t` needs to flushed.